docker-compose.yml

```diff
@@ -0,0 +1,55 @@
+version: "3.8"
+services:
+  devel:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        ROS_DISTRO: $ROS_DISTRO
+    image: oriserobotics/ros-$ROS_DISTRO:devel
+    runtime: nvidia
+    environment:
+      - NVIDIA_VISIBLE_DEVICES=all
+      - NVIDIA_DRIVER_CAPABILITIES=all
+      - DISPLAY
+      - COLCON_WORKSPACE_FOLDER=$COLCON_WORKSPACE_FOLDER
+      - USER=orise
+    network_mode: host
+    volumes:
+      - ~/.ssh/known_hosts:/home/orise/.ssh/known_hosts:rw
+      - ~/.ssh/id_rsa:/home/orise/.ssh/id_rsa:ro
+      - ~/.ssh/id_rsa.pub:/home/orise/.ssh/id_rsa.pub:ro
+      - ~/.gitconfig:/etc/gitconfig:ro
+      - /tmp/.X11-unix:/tmp/.X11-unix:rw
+      - $VOLUMES_FOLDER/$CONTAINER_NAME:/home/orise:rw
+      - /etc/localtime:/etc/localtime:ro
+    privileged: true
+    working_dir: $COLCON_WORKSPACE_FOLDER
+    container_name: $CONTAINER_NAME
+    stdin_open: true # docker run -i
+    tty: true # docker run -t
+  test:
+    build:
+      context: .
+      dockerfile: Dockerfile.test
+      args:
+        ROS_DISTRO: $ROS_DISTRO
+    runtime: nvidia
```

Comment:
I've just realized this is a feature of spec v2 that is valid in the latest format supported since. It is marked as deprecated because it is hardware-specific. Therefore, I think I should remove it from here and move these definitions to the script as optional configurations

Reply:
What do you mean to add in a script as option? Remove docker compose?

Reply:
My bad. `docker-compose` does not offer a CLI option for it.

Reply:
I found that v 1.28 introduced [Support for Nvidia GPUs via device requests](https://github.com/docker/compose/releases). This seems to be the most appropriate long-term solution.

I'm not sure I can make docker-compose work properly with nvidia for versions older than 1.27 (considering the runtime option) or 1.28 (consider the new feature). If I confirm that, I suggest 2 options:

1. Do not offer nvidia support
2. State this feature demands manual installation of a newer version of docker-compose

We can also keep the older with `docker run`

Reply:
I vote on option 2

Reply:
Option 2 adopted: b2f0946